src/main/java/org/gridsuite/voltageinit/server/computation/service/AbstractWorkerService.java

```diff
@@ -0,0 +1,241 @@
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Consumer;
+
+/**
+ * @author Mathieu Deharbe <mathieu.deharbe at rte-france.com>
+ * @param <S> powsybl Result class specific to the computation
+ * @param <R> Run context specific to a computation, including parameters
+ * @param <P> powsybl and gridsuite Parameters specifics to the computation
+ * @param <T> result service specific to the computation
+ */
+public abstract class AbstractWorkerService<S, R extends AbstractComputationRunContext<P>, P, T extends AbstractComputationResultService<?>> {
+    protected static final Logger LOGGER = LoggerFactory.getLogger(AbstractWorkerService.class);
+
+    protected final Lock lockRunAndCancel = new ReentrantLock();
+    protected final ObjectMapper objectMapper;
+    protected final NetworkStoreService networkStoreService;
+    protected final ReportService reportService;
+    protected final ExecutionService executionService;
+    protected final NotificationService notificationService;
+    protected final AbstractComputationObserver<S, P> observer;
+    protected final Map<UUID, CompletableFuture<S>> futures = new ConcurrentHashMap<>();
+    protected final Map<UUID, CancelContext> cancelComputationRequests = new ConcurrentHashMap<>();
+    protected final T resultService;
+
+    protected AbstractWorkerService(NetworkStoreService networkStoreService,
+                                    NotificationService notificationService,
+                                    ReportService reportService,
+                                    T resultService,
+                                    ExecutionService executionService,
+                                    AbstractComputationObserver<S, P> observer,
+                                    ObjectMapper objectMapper) {
+        this.networkStoreService = networkStoreService;
+        this.notificationService = notificationService;
+        this.reportService = reportService;
+        this.resultService = resultService;
+        this.executionService = executionService;
+        this.observer = observer;
+        this.objectMapper = objectMapper;
+    }
+
+    protected PreloadingStrategy getNetworkPreloadingStrategy() {
+        return PreloadingStrategy.COLLECTION;
+    }
+
+    protected Network getNetwork(UUID networkUuid, String variantId) {
+        Network network;
+        try {
+            network = networkStoreService.getNetwork(networkUuid, getNetworkPreloadingStrategy());
+            String variant = StringUtils.isBlank(variantId) ? VariantManagerConstants.INITIAL_VARIANT_ID : variantId;
+            network.getVariantManager().setWorkingVariant(variant);
+        } catch (PowsyblException e) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, e.getMessage());
+        }
+        return network;
```

Comment:
Don't need variable outside of try-catch scope.
```suggestion
        try {
            Network network = networkStoreService.getNetwork(networkUuid, getNetworkPreloadingStrategy());
            String variant = StringUtils.isBlank(variantId) ? VariantManagerConstants.INITIAL_VARIANT_ID : variantId;
            network.getVariantManager().setWorkingVariant(variant);
            return network;
        } catch (PowsyblException e) {
            throw new ResponseStatusException(HttpStatus.NOT_FOUND, e.getMessage());
        }
```